sample-code/javascript-wd/test/basic/android-basic-interactions.test.js

```diff
@@ -0,0 +1,63 @@
+import wd from 'wd';
+import chai from 'chai';
+import { androidCaps, serverConfig } from '../helpers/caps';
+
+const {assert} = chai;
+
+const PACKAGE = 'io.appium.android.apis';
+const SEARCH_ACTIVITY = '.app.SearchInvoke';
+const ALERT_DIALOG_ACTIVITY = '.app.AlertDialogSamples';
+
+describe('Basic Android interactions', function () {
+
+  let driver;
+
+  before(async function () {
+    // Connect to Appium server
+    driver = await wd.promiseChainRemote(serverConfig);
+
+    // Start the session
+    await driver.init({
+      ...androidCaps,
+      app: require('../helpers/apps').androidApiDemos,
+      appActivity: SEARCH_ACTIVITY, // Android-specific capability. Can open a specific activity.
+    });
+  });
+
+  after(async function () {
+    await driver.quit();
+  });
+
+  it('should send keys to search box and then check the value', async function () {
+    // Enter text in a search box
+    const searchBoxElement = await driver.elementById('txt_query_prefill');
+    await searchBoxElement.sendKeys('Hello world!');
+
+    // Press on 'onSearchRequestedButton'
+    const onSearchRequestedButton = await driver.elementById('btn_start_search');
+    await onSearchRequestedButton.click();
+
+    // Check that the text matches the search term
+    await driver.waitForElementById('android:id/search_src_text');
+    const searchText = await driver.elementById('android:id/search_src_text');
+    const searchTextValue = await searchText.text();
+    assert.equal(searchTextValue, 'Hello world!');
+  });
+
+  it('should click a button that opens an alert and then dismisses it', async function () {
+    // Open the 'Alert Dialog' activity of the android app
+    await driver.startActivity({appPackage: PACKAGE, appActivity: ALERT_DIALOG_ACTIVITY});
+
+    // Click button that opens a dialog
+    const openDialogButton = await driver.elementById('io.appium.android.apis:id/two_buttons');
+    await openDialogButton.click();
+    await driver.waitForElementById('android:id/alertTitle');
+
+    // Check that the dialog is there
+    const alertText = (await driver.elementById('android:id/alertTitle')).text();
+    const closeDialogButton = await driver.elementById('android:id/button1');
```

Comment:
Do the normal alert methods not work with UIAutomator2?
```js
const alertText = await driver.alertText();
await driver.dismissAlert();
```

Reply:
Apparently not. This is what happens:

```
[debug] [JSONWP Proxy] Proxying [POST /wd/hub/session/7122ce0b-51ad-40cd-901d-a1ed14ef9cc3/dismiss_alert] to [POST http://localhost:8200/wd/hub/session/6423b5af-e40c-4a78-857d-ec55caaf6699/dismiss_alert] with body: {}
[MJSONWP] Encountered internal error running command: Error: Could not proxy. Proxy error: Could not proxy command to remote server. Original error: 404 - undefined
    at doJwpProxy$ (/Users/danielgraham/appium/node_modules/appium-base-driver/lib/mjsonwp/mjsonwp.js:408:13)
    at tryCatch (/Users/danielgraham/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/danielgraham/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Users/danielgraham/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Users/danielgraham/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
```